script/test-utils.ts

```diff
@@ -0,0 +1,53 @@
+import path from 'node:path';
+import fs from 'node:fs';
+import { tmpdir } from 'node:os';
+import { execSync } from 'node:child_process';
+
+export const PNPM_STORE_PATH = path.resolve(__dirname, '../.pnpm-test-store');
+export const NPM_RC_FILE = '.npmrc';
+export const NPM_RC_CONTENTS = `store-dir = ${PNPM_STORE_PATH}`;
+export const PACKAGE_JSON_FILE = 'package.json';
+export const PACKAGE_JSON_CONTENTS = '{"name":"test-project","version":"1.0.0"}';
+
+export function preparePackageJson(dependencies: {[key: string]: string} = {}, devDependencies: {[key: string]: string} = {}): string {
+  const tmpDir = fs.mkdtempSync(path.join(tmpdir(), 'zenstack-test-'));
```

Comment:
Given that this is a loose file included from elsewhere, I couldn't rely on the tmp package here and had to go with built-in node functions. Any downsides to this that I'm not considering?

(My thinking for including it this way was the developer experience of not needing to do a build step inbetween. Thoughts on this?)